### DISCUSSION TITLE
HOW to FLASH ESP32-S3-DevKitC-1-N32R8V

### DISCUSSION THREAD

**Discussion by martinsamus** [emsesp/EMS-ESP32#1635] [General]
Hi everyone,

Has anyone tried to flash firmware to ESP32-S3-DevKitC-1-N32R8V. Is ESP32-S3-DevKitC-1-N32R8V supported?
If so, how can I flash it?

well thank you

**Comment by proddy** [comment]
have you tried the EMS-ESP Flasher Tool yet? https://emsesp.github.io/docs/Getting-Started/#first-time-install

**Reply by martinsamus** [reply]
Yes, I tried it, just like the first option:
![image](https://github.com/emsesp/EMS-ESP32/assets/11224829/d6d2f83c-90d8-4b78-b5db-548f24e45aee)
![image](https://github.com/emsesp/EMS-ESP32/assets/11224829/e3662613-f94e-42bb-aa99-dfc601051d5f)
After flashing, a boot loop occurred with error: SHA-256 comparison failed

Note: The ESP32-S3 was detected with 4 MB flash, but it has 32MB flash and 8MB SPRAM.

**Reply by MichaelDvP** [reply]
I think best way is to use platformio upload. Then you can create a partition scheme for the 32M and use the full chip.

**Reply by martinsamus** [reply]
I have never used platformio before. Can you please write steps how to do it? Thanks

**Reply by proddy** [reply]
@MichaelDvP shall we make a `esp32_partition_32M.csv` like:

```
# Name,   Type, SubType, Offset,   Size,    Flags
nvs,      data, nvs,     0x9000,   0x5000,
otadata,  data, ota,     ,         0x2000,
app0,     app,  ota_0,   ,         0xFE0000,
app1,     app,  ota_1,   ,         0xFE0000,
spiffs,   data, spiffs,  ,         64K,
```

and a new env in `platformio.ini` with:

```ini
[env:ci_s3_32M]
extends = espressi32_base
extra_scripts = scripts/rename_fw.py
board = lolin_s3
board_build.f_cpu = 240000000L
board_upload.flash_size = 32MB
board_build.partitions = esp32_partition_32M.csv
build_unflags = ${common.unbuild_flags}
build_flags =
  ${espressi32_base.build_flags}
  '-DEMSESP_DEFAULT_BOARD_PROFILE="S32S3"'
```

**Reply by martinsamus** [reply]
Thank for configuration, but something is wrong (builded by Visual Studio Code and uploaded by platformio):
![obrázok](https://github.com/emsesp/EMS-ESP32/assets/11224829/7504186c-8d10-4289-9afd-34ac3d0b55c4)
![obrázok](https://github.com/emsesp/EMS-ESP32/assets/11224829/2b7f4205-e08b-41c4-80d4-31e74b3ad402)

I don't understand one thing. When I flashed using esptool, I needed bootloader.bin, partitions.bin, boot_app0.bin and firmware.bin. Now, when I used platformio, only the new firmware EMS-ESP-3_6_4-ESP32_S3_32M.bin was built. What bin files was used in the upload as bootloader, partition and boot_app0???
Well thank you.

**Reply by MichaelDvP** [reply]
Maybe you need to test a bit, the board is new and not in platformio, also not compatible with the lolin-s3. https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html shows it uses WROOM2 with octal spi flash. I can not find a predefined board with this module. 
Try to add to the platformio.ini env::ci_s3_32M: `board_build.flash_mode = opi`
I don't know if we can overwrite the arduino.memory_type, try with `board_build.arduino.memory_type: opi_opi`
Before flash use in VSC->platformio->ci_s3_32M->platform: Erase Flash to make sure there is nothing left from early tests.

Or go to users directoy `.platformio\platforms\espressif32\boards` use lolin-s3.json as base and add a new  board with the flash settings, then you can use it in the project environment.

For the partition table you can start with proddys suggestion, but for later it's better add moe data space, you'll never use the space of the large app partitions. I suggest also to add coredump:
```
# Name,   Type, SubType, Offset,   Size,    Flags
nvs,      data, nvs,     0x9000,   0x005000,
otadata,  data, ota,     ,         0x002000,
app0,     app,  ota_0,   ,         0xDD0000,
app1,     app,  ota_1,   ,         0xDD0000,
nvs1,     data, nvs,     ,         0x040000,
spiffs,   data, spiffs,  ,         0x400000,
coredump, data, coredump,,         0x010000,
```

The firmware.bin, bootloader.bin and partitions.bin are generated on compile and stored in project_directory `.pio/build/ci_s3_32M`

**Reply by martinsamus** [reply]
Thanks, it worked, a bit of testing and here is the result:
![obrázok](https://github.com/emsesp/EMS-ESP32/assets/11224829/9555f6b2-adfd-46e9-819a-7546343eb2ce)
![obrázok](https://github.com/emsesp/EMS-ESP32/assets/11224829/13896393-1ba0-4b50-9546-24fdcc506bb8)

My configuration for others:
edit file platformio.ini:
```
[platformio]
default_envs = ci_s3_32M

[env:ci_s3_32M]
extends = espressi32_base
extra_scripts = scripts/rename_fw.py
board = lolin_s3_32
board_build.f_cpu = 240000000L
board_build.flash_mode = opi
board_build.arduino.memory_type = opi_opi
board_upload.flash_size = 32MB
board_build.partitions = esp32_partition_32M.csv
build_unflags = ${common.unbuild_flags}
build_flags =
  ${espressi32_base.build_flags}
  '-DEMSESP_DEFAULT_BOARD_PROFILE="S32S3"'
```

create file in users directory/.platformio/platforms/espressif32/boards/lolin_s3_32.json:
```
{
  "build": {
    "arduino": {
      "ldscript": "esp32s3_out.ld",
      "partitions": "default_32MB.csv",
      "memory_type": "opi_opi"
    },
    "core": "esp32",
    "extra_flags": [
      "-DBOARD_HAS_PSRAM",
      "-DARDUINO_LOLIN_S3",
      "-DARDUINO_USB_MODE=1",
      "-DARDUINO_USB_CDC_ON_BOOT=1"
    ],
    "f_cpu": "240000000L",
    "f_flash": "80000000L",
    "flash_mode": "opi",
    "hwids": [
      [
        "0x303A",
        "0x1001"
      ]
    ],
    "mcu": "esp32s3",
    "variant": "lolin_s3"
  },
  "connectivity": [
    "wifi"
  ],
  "debug": {
    "openocd_target": "esp32s3.cfg"
  },
  "frameworks": [
    "arduino",
    "espidf"
  ],
  "name": "WEMOS LOLIN S3",
  "upload": {
    "flash_size": "32MB",
    "maximum_ram_size": 327680,
    "maximum_size": 16777216,    
    "use_1200bps_touch": true,
    "wait_for_upload_port": true,
    "require_upload_port": true,
    "speed": 460800
  },
  "url": "https://www.wemos.cc/en/latest/s3/index.html",
  "vendor": "WEMOS"
}
```

create file in users directory/.platformio/packages/framework-arduinoespressif32/tools/partitions/default_32MB.csv:
```
# Name,   Type, SubType, Offset,   Size,    Flags
nvs,      data, nvs,     0x9000,   0x005000,
otadata,  data, ota,     ,         0x002000,
app0,     app,  ota_0,   ,         0xDD0000,
app1,     app,  ota_1,   ,         0xDD0000,
nvs1,     data, nvs,     ,         0x040000,
spiffs,   data, spiffs,  ,         0x400000,
coredump, data, coredump,,         0x010000,
```
Note: esp32_partition_32M.csv has same content as default_32MB.csv

Many thanks to @MichaelDvP and @proddy
